src/app/services/dialog.service.ts

```diff
@@ -77,13 +77,16 @@ export class DialogService {
       dialogRef.componentInstance.method = options.method;
       dialogRef.componentInstance.switchSelectionEmitter.pipe(untilDestroyed(this)).subscribe((selection: boolean) => {
         const data = options.data;
-        if (selection) {
+        if (selection && data[0]) {
```

Comment:
Looks weird. It's good to revert it.

Reply:
Without the check `data[0]`, I have to pass an object in the dialog config and introduce new ifs into the service with special code just for this PR. This check helps me avoid that scenario. I can just subscribe to the selection change emitter and do my own thing.